langroid/agent/base.py

```diff
@@ -58,6 +70,7 @@ def __init__(self, config: AgentConfig):
         self.config = config
         self.dialog: List[Tuple[str, str]] = []  # seq of LLM (prompt, response) tuples
         self.llm_tools_map: Dict[str, Type[ToolMessage]] = {}
+        self.all_llm_responses: List[LLMResponse] = []
```

Comment:
we don't need to store the entire LLM responses. We are only using this to compute total cost. Instead have a `self.llm_token_cost = 0`, and accumulate this whenever there is a (non-cached) LLM response.

Reply:
If we want to accumulate each component of the cost/usage, we can have this member variable `self.llm_token_cost` be of  `TokenCost` defined in the `language_model.base`.

Reply:
I think you deduced my reasoning for collecting LLM responses. I don't want only the final accumulated cost, but also the cost of individual LLM responses. 
bwt, that list `self.all_llm_responses` maintains only non-cached responses.

Reply:
yes but no need to store the actual responses. Just have a variable of type `TokenCost` that accumulates the various components, each time there is a non-cached LLM response

Reply:
> If we want to accumulate each component of the cost/usage, we can have this member variable `self.llm_token_cost` be of `TokenCost` defined in the `language_model.base`.

do you mean something like this under `LLMTokenUsage`?
```
@property
    def llm_token_cost(self) -> float:
        return self.prompt_tokens*inputCost + self.completion_tokens*outputCost
```

Reply:
because this means I need to have a member `config: OpenAIGPTConfig` under `LLMTokenUsage` to get the cost

Reply:
no I just meant use `LLMTokenUsage` instead of the dict, rest stays same, except you also have `total_tokens` be a property that just sums the two components.

Reply:
I think you would use this `LLMTokenUsage` in a couple places